src/main/java/org/semux/consensus/SemuxBft.java

```diff
@@ -790,9 +802,17 @@ protected Block proposeBlock() {
          */
         timestamp = timestamp > parent.getTimestamp() ? timestamp : parent.getTimestamp() + 1;
 
-        // signal UNIFORM_DISTRIBUTION fork
-        byte[] data = signalingUniformDistribution()
-                ? BlockHeaderData.v1(new BlockHeaderData.ForkSignalSet(UNIFORM_DISTRIBUTION)).toBytes()
+        // signal forks
+        Set<ValidatorActivatedFork> forks = new HashSet<>();
+        if (signalingUniformDistribution()) {
+            forks.add(UNIFORM_DISTRIBUTION);
+        }
+        if (signalingUniformDistribution()) {
+            forks.add(VIRTUAL_MACHINE);
+        }
+        byte[] data = !forks.isEmpty()
+                ? BlockHeaderData.v1(new BlockHeaderData.ForkSignalSet(
+                        forks.toArray(new ValidatorActivatedFork[forks.size()]))).toBytes()
```

Comment:
Looks good, but we should consider abstract the signaling into a method, and add unit test.